types/node/tls.d.ts

```diff
@@ -998,7 +998,7 @@ declare module "tls" {
      * @param hostname The host name or IP address to verify the certificate against.
      * @param cert A `certificate object` representing the peer's certificate.
      */
-    function checkServerIdentity(hostname: string, cert: PeerCertificate): Error | undefined;
```

Comment:
Should this be only `cert: DetailedPeerCertificate` (non union) like the other cases?

Reply:
The union shouldn't be necessary because `DetailedPeerCertificate` is a subtype of `PeerCertificate`, so it is already a legal argument:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b69f27d2c5707ff5f14fae100693443af0caa932/types/node/tls.d.ts#L133